controllers/topology/reconcile_state.go

```diff
@@ -400,6 +517,13 @@ func (r *ClusterReconciler) deleteMachineDeployment(ctx context.Context, cluster
 		return errors.Wrapf(err, "failed to delete %s", tlog.KObj{Obj: md.Object})
 	}
 	r.recorder.Eventf(cluster, corev1.EventTypeNormal, deleteEventReason, "Deleted %q", tlog.KObj{Obj: md.Object})
+
+	// Patch MachineHealthCheck for the MachineDeployment.
+	if md.MachineHealthCheck != nil {
```

Comment:
Q: Why do we delete the MHC here and in the MD controller?

(Note: if the MHC deletion fails here this code is not reentrant, but that's probably not important as the MD controller deletes the MD too)

Reply:
Okay thx. I think deleting the MHC before the MD could do the trick.